command.php

```diff
@@ -348,6 +338,64 @@ public function is_woocommerce_blocks_active(): bool {
 
 		return 0 === $result->return_code;
 	}
+
+	/**
+	 * Installs WooCommerce Blocks plugin.
+	 *
+	 * @param array $assoc_args An array with optional arguments.
+	 *
+	 * @return void
+	 */
+	private function installWoocommerceBlocksPlugin( array $assoc_args ) {
+		if ( ! isset( $assoc_args['blocks'] ) ) {
+			return;
+		}
+
+		if ( true === $assoc_args['blocks'] ) {
+			WP_CLI::runcommand( 'plugin install woo-gutenberg-products-block --activate' );
+			return;
+		}
+
+		$url = '';
+		if ( $this->isReleaseVersion( $assoc_args['blocks'] ) ) {
+			$url = "https://github.com/woocommerce/woocommerce-gutenberg-products-block/releases/download/v{$assoc_args['blocks']}/woo-gutenberg-products-block.zip";
+		} elseif ( $this->isUrl( $assoc_args['blocks'] ) ) {
+			$url = $assoc_args['blocks'];
+		}
+
+		if ( ! $url ) {
+			return;
+		}
+
+		try {
+			WP_CLI::runcommand( "plugin install {$url} --activate" );
+		} catch ( \Throwable $th ) {
+			WP_CLI::error( "WooCommerce Blocks release {$assoc_args['block']} could not be installed!" );
+			WP_CLI::error( $th );
+		}
+	}
+
+	/**
+	 * Checks if the given string is a release version.
+	 *
+	 * @param string $version The version to check.
+	 *
+	 * @return false|int
+	 */
+	public function isReleaseVersion( $version ) {
+		return preg_match( '/^\d.\d.\d$/', $version );
+	}
+
+	/**
+	 * Checks if the given string is a URL.
+	 *
+	 * @param string $url The URL to check.
+	 *
+	 * @return bool
+	 */
+	public function isUrl( $url ): bool {
+		return filter_var( $url, FILTER_VALIDATE_URL );
+	}
```

Comment:
Very well done! I really like how you refactored this part and elegant the code looks now! 👏